AdyenCheckout/AdyenCheckout.swift

```diff
@@ -9,11 +9,39 @@
 @_spi(AdyenInternal) import AdyenSession
 @_spi(AdyenInternal) import AdyenNetworking
 
-public class AdyenCheckout {
+internal protocol AdyenCheckoutProtocol {
+    static func setup(
+        with sessionId: String,
+        sessionData: String,
+        configuration: CheckoutConfiguration,
+        completion: @escaping (Result<AdyenCheckout, Error>) -> Void
+    )
+    
+    static func setup(
+        with paymentMethods: PaymentMethods,
+        configuration: CheckoutConfiguration,
+        completion: @escaping (Result<AdyenCheckout, Error>) -> Void
+    )
+    
+    static func setupSession(
+        with configuration: CheckoutConfiguration,
+        order: PartialPaymentOrder?,
+        completion: @escaping (Result<AdyenSession, Error>) -> Void
+    )
+    
+    static func fetchCheckoutAttemptId(
```

Comment:
I'm in doubt this belongs here. Let's apply interface segregation and introduce `CheckoutAttemptIdProviding`?